### PR TITLE
Update config and run Detekt during the build

### PR DESCRIPTION
**Pull request by yevhenii-nadtochii**
This PR does the following:

1. Updates to the latest `config`.
2. Utilizes `Spine` dependency object.
3. Applies Detekt to modules.

Important notes:
1. Local `Protobuf.kt` uses a higher version (`3.21.6`) compared to `config` (`3.19.6`). The local one is kept.
3. I've preserved the declaration of the used versions of Spine modules in `version.gradle.kts`. It's needed to adjust the code in this module to address changes made in `core-java` when the versions from `Spine` are used. I can do it in a separate PR. 

The library version is bumped to `2.0.0-SNAPSHOT.121`.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -92,6 +97,11 @@ all modules and does not describe the project structure per-subproject.
     <artifactId>checkstyle</artifactId>
     <version>10.3.4</version>
   </dependency>
+  <dependency>
+    <groupId>io.gitlab.arturbosch.detekt</groupId>
+    <artifactId>detekt-cli</artifactId>
+    <version>1.21.0</version>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:103]: ignore

File: pom.xml [hunk 2]
```diff
@@ -74,13 +74,18 @@ all modules and does not describe the project structure per-subproject.
...
     <artifactId>junit-jupiter-engine</artifactId>
-    <version>5.8.2</version>
+    <version>5.9.1</version>
     <scope>test</scope>
   </dependency>
   <dependency>
     <groupId>com.google.errorprone</groupId>
     <artifactId>error_prone_core</artifactId>
-    <version>2.13.1</version>
+    <version>2.16</version>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:83]: ignore

**Comment by yevhenii-nadtochii** [comment]
@alexander-yevsyukov @armiol Please take a look.

**Comment by armiol** [comment]
@yevhenii-nadtochii 

1. Please keep the lower version of Protobuf here. This is because the newer one has its issues.
2. Separate PR is fine.

File: buildSrc/src/main/kotlin/io/spine/internal/dependency/Protobuf.kt [hunk 3]
```diff
@@ -30,7 +30,7 @@ package io.spine.internal.dependency
 @Suppress("MemberVisibilityCanBePrivate") // used directly from outside
 object Protobuf {
     private const val group = "com.google.protobuf"
-    const val version       = "3.21.6"
+    const val version       = "3.21.9"
```
Comment (armiol) [inline buildSrc/src/main/kotlin/io/spine/internal/dependency/Protobuf.kt]: Please see my standalone comment on this matter.

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,5 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions-snapshots/gradle-7.6-20220930142900+0000-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.6-rc-1-bin.zip
```
Comment (armiol) [inline gradle/wrapper/gradle-wrapper.properties]: I think we now have `...rc-2`. Please review.

**Review by armiol — APPROVED** [review]
@yevhenii-nadtochii LGTM with some comments to address.

**Pull request merged**